browser/profiles/profile_util.cc

```diff
@@ -215,6 +220,22 @@ bool IsRegularProfile(content::BrowserContext* context) {
          profile->IsRegularProfile();
 }
 
+void RecordInitialP3AValues(Profile* profile) {
+  // Preference is unregistered for some reason in profile_manager_unittest
+  // TODO(bsclifton): do a better fix
+  if (!profile->GetPrefs()->FindPreference(kNewTabPageShowBackgroundImage) ||
+      !profile->GetPrefs()->FindPreference(
+          kNewTabPageShowSponsoredImagesBackgroundImage)) {
+    return;
+  }
```

Comment:
RE: the above, I wasn't sure of a better fix for unit test path. When testing profile creation, this ends up firing and prefs are undefined in the unit test. Let me know if this is not acceptable

Reply:
I think this checking seems fine. Not all unittest need to register whole prefs.

Reply:
we probably should've instantiate testing profile in unit tests